tests/arch/arm/arm_thread_swap/src/arm_syscalls.c

```diff
@@ -241,3 +242,60 @@ void test_arm_syscalls(void)
 /**
  * @}
  */
+
+
+void z_impl_test_arm_cpu_write_reg(void)
+{
+	/* User thread CPU write registers system call for testing
+	 *
+	 * Verify the following
+	 * - Write 0xDEADBEEF values during system call into registers
+	 * - In main test we will read that registers to verify
+	 * that all of them were scrubbed and do not contain any sensitive data
+	 */
+
+	/* Part below is made to test that kernel scrubs CPU registers
+	 * after returning from the system call
+	 */
+	__asm__ volatile (
+		"ldr r0, =0xDEADBEEF;\n\t"
+		"ldr r1, =0xDEADBEEF;\n\t"
+		"ldr r2, =0xDEADBEEF;\n\t"
+		"ldr r3, =0xDEADBEEF;\n\t"
+		);
+}
+
+static inline void z_vrfy_test_arm_cpu_write_reg(void)
+{
+	z_impl_test_arm_cpu_write_reg();
+}
+#include <syscalls/test_arm_cpu_write_reg_mrsh.c>
+
+/**
+ * @brief Test CPU scrubs registers after system call
+ *
+ * @details - Call from user mode a syscall test_arm_cpu_write_reg(),
+ * the system call function writes into registers 0xDEADBEEF value
+ * - Then in main test function below check registers values,
+ * if no 0xDEADBEEF value detected, that means CPU scrubbed registers
+ * before exit from the system call.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_syscall_cpu_scrubs_regs(void)
```

Comment:
@maksimmasalski i suppose the test should be skipped if CONFIG_USERSPACE is not defined.

Reply:
I added that test in proper branch in main.c file